revocation/result/results.go

```diff
@@ -0,0 +1,80 @@
+// Package result provides general objects that are used across revocation
+package result
+
+import "strconv"
+
+// Result is a type of enumerated value to help characterize errors. It can be
+// OK, Unknown, or Revoked
+type Result int
+
+const (
+	// ResultOK is a Result that indicates that the revocation check resulted in no
+	// important errors
+	ResultOK Result = 1 + iota
+	// ResultNonRevokable is a Result that indicates that the certificate cannot be
+	// checked for revocation. This may be a result of no OCSP servers being
+	// specified, the cert is a root certificate, or other related situations.
+	ResultNonRevokable
+	// ResultUnknown is a Result that indicates that some error other than a
+	// revocation was encountered during the revocation check
+	ResultUnknown
+	// ResultRevoked is a Result that indicates that at least one certificate was
+	// revoked when performing a revocation check on the certificate chain
+	ResultRevoked
+)
```

Comment:
nit: you can put `ResultUnknown` as the first one so that default value is unknown.
```suggestion
const (
	// ResultUnknown is a Result that indicates that some error other than a
	// revocation was encountered during the revocation check
	ResultUnknown = iota
	// ResultOK is a Result that indicates that the revocation check resulted in no
	// important errors
	ResultOK Result
	// ResultNonRevokable is a Result that indicates that the certificate cannot be
	// checked for revocation. This may be a result of no OCSP servers being
	// specified, the cert is a root certificate, or other related situations.
	ResultNonRevokable
	// ResultRevoked is a Result that indicates that at least one certificate was
	// revoked when performing a revocation check on the certificate chain
	ResultRevoked
)
```